src/3box.js

```diff
@@ -117,30 +117,40 @@ class ThreeBox {
       // console.log(this._rootStore.iterator({ limit: -1 }).collect().length)
       // console.log(await this._ipfs.pubsub.peers('/orbitdb/QmRxUAGk62v7NjUkzvcqwYkBqF3zHb8tfhfW6T3MateGje/b932fe7ab.root'))
       console.time('replicate rootStore')
-      if (!this._rootStore.iterator({ limit: -1 }).collect().length) {
+      const entries = this._rootStore.iterator({ limit: -1 }).collect()
+      const rootStoreEmpty = !entries.length
+      // const rs = this._rootStore
+      // function replicatedRootStore () {
+      //   return new Promise((resolve, reject) => {
+      //     rs.events.on('replicated', () => {
```

Comment:
The `'replicated'` event is emitted when an entry has been received **and** fully added to the local odb.

Reply:
I asked haad about this. I though this would only be emitted when the entire store had been replicated, but turns out it's emitted for every event.